File: Sorting/mergeSort.cpp
#include <bits/stdc++.h>
using namespace std;
void mergeSort(int arr1[], int n, int arr2[], int m, int arr3[])
{ // n= size of arr1, m= size of arr2
    int i = 0, j = 0, k = 0;
    while (i <= n && j <= m)
    {
        if (arr1[i] < arr2[j])
        {
            arr3[k++] = arr1[i++];
            // k++;
            // i++;
        }
        else
        {
            arr3[k++] = arr2[j++];
            // k++;
            // j++;
        }
    }
    while (i < n)
    {
        arr3[k++] = arr1[i++];
        // k++;
        // i++;
    }
    while (j < m)
    {
        arr3[k++] = arr2[j++];
        // k++;
        // j++;
    }
}

// or//
/*
void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        // Here, nums1 =1st array, nums2 = 2nd array...
        for(int i = 0 ; i < n ; i++)
            nums1[i + m] = nums2[i];
            // Sort the 1st array...
            sort(nums1.begin() , nums1.end());
        // Print the required result...
        return;
    }
*/
int main()
{
    int arr1[6] = {1, 5, 7, 10, 13, 16};
    int arr2[4] = {2, 3, 6, 11};
    int arr3[11] = {0};
    mergeSort(arr1, 6, arr2, 3, arr3);
    for (int i = 0; i < 10; i++)
    {
        cout << arr3[i] << " ";
    }
    return 0;
}

File: Sorting/mergeSort.py
def mergeSort(arr1,n,arr2,m,arr3):
    i=0
    j=0
    # k=0
    while(i<=n and j<=m):
        if(arr1[i]<arr2[j]):
            arr3.append(arr1[i])
            # k+=1
            i+=1
        else:
            arr3.append(arr2[j])
            # k+=1
            j+=1
    while(i<n):
        arr3.append(arr1[i])
        # k+=1
        i+=1
    while(j<m):
        arr3.append(arr2[j])
        # k+=1
        j+=1
arr1=[1,3,5,7,9,10,11]
arr2=[2,6,10]
arr3=[]
mergeSort(arr1,len(arr1),arr2,len(arr2)-1,arr3)
print(arr3)